src/consensus/pbft/libbyz/Node.cpp

```diff
@@ -116,16 +115,17 @@ void Node::add_principal(const PrincipalInfo& principal_info)
   {
     LOG_INFO << "Principal with id: " << principal_info.id
              << " has already been configured" << std::endl;
+    auto principal = it->second;
```

Comment:
IIUI, we replace the certificate of an existing principal with the certificate of the new "duplicated" (based on id) principal, but nothing else. It seems a little odd. Why is this required?

Reply:
The principal object will be setup during the first `Status` message (see `Status::pre_verify()`) but the request will be empty. We will get the certificate for this principal when `add_node` is triggered via the `node`s table hook so we want to set the certificate there.